<launch>
    <arg name="namespace" default="taluy" />
    <arg name="rate" default="20.0" />

    <!-- Engine file path (required) -->
    <arg name="engine_path" default="$(find auv_detection)/models/DA3METRIC-LARGE_476x644_sim.engine" />

    <arg name="camera_namespace" default="cameras/cam_front" />
    <arg name="input_image_topic" default="$(arg camera_namespace)/image_rect_color" />
    <arg name="output_depth_topic" default="depth_anything/raw_depth" />
    <arg name="output_colorized_topic" default="depth_anything/colorized" />
    <arg name="scaled_camera_info_topic" default="depth_anything/scaled_camera_info" />
    <arg name="output_points_topic" default="depth_anything/points" />

    <group ns="$(arg namespace)">
        <node pkg="auv_vision" type="depth_anything_trt_node.py" name="depth_anything_trt_node" output="screen">
            <param name="engine_path" value="$(arg engine_path)" />
            <param name="rate" value="$(arg rate)" />
            <param name="camera_namespace" value="$(arg camera_namespace)" />

            <remap from="image_raw" to="$(arg input_image_topic)" />
            <remap from="raw_depth" to="$(arg output_depth_topic)" />
            <remap from="colorized" to="$(arg output_colorized_topic)" />
            <remap from="scaled_camera_info" to="$(arg scaled_camera_info_topic)" />
            <remap from="points" to="$(arg output_points_topic)" />
        </node>
    </group>
</launch>
